metrics/stackdriver_exporter.go

```diff
@@ -115,3 +160,44 @@ func getMetricTypeFunc(metricTypePrefix, customMetricTypePrefix string) func(vie
 		return path.Join(customMetricTypePrefix, view.Measure.Name())
 	}
 }
+
+// getStackdriverSecret returns the Kubernetes Secret specified in the given config.
+func getStackdriverSecret(sdconfig *stackdriverClientConfig) (*corev1.Secret, error) {
+	err := ensureKubeclient()
+	if err != nil {
```

Comment:
Given that `ensureKubeclient` seems to be side-effect only, why not collapse like so:

```golang
if err := ensureKubeclient(); err != nil {
  ...
}
```

Reply:
Pending